### PR TITLE
gh-104773: PEP 594: Remove the telnetlib module

### PR DESCRIPTION
**Pull request by vstinner**
* Issue: gh-104773

----
:books: Documentation preview :books:: https://cpython-previews--104778.org.readthedocs.build/

### CODE REVIEW

**Review by gpshead — APPROVED** [review]

File: Doc/whatsnew/3.12.rst [hunk 1]
```diff
@@ -819,7 +819,7 @@ Modules (see :pep:`594`):
 * :mod:`sndhdr`
 * :mod:`spwd`
 * :mod:`sunau`
-* :mod:`telnetlib`
+* ``telnetlib``
```
Comment (hugovk) [inline Doc/whatsnew/3.12.rst]: Optionally:
```suggestion
* :mod:`!telnetlib`
```

File: Doc/whatsnew/3.13.rst [hunk 2]
```diff
@@ -102,6 +102,10 @@ Deprecated
 Removed
 =======
 
+* Remove the ``telnetlib`` module, deprecated in Python 3.11: use the projects
+  `telnetlib3 <https://pypi.org/project/telnetlib3/>`_
+  or `Exscript <https://pypi.org/project/Exscript/>`_ instead.
+  (Contributed by Victor Stinner in :gh:`104773`.)
```
Comment (hugovk) [inline Doc/whatsnew/3.13.rst:108]: Let's add a reference to the PEP.

File: Misc/NEWS.d/next/Library/2023-05-23-02-20-13.gh-issue-104773.7K59zr.rst
Comment (hugovk) [inline Misc/NEWS.d/next/Library/2023-05-23-02-20-13.gh-issue-104773.7K59zr.rst:1]: Shall we reference the PEP here too?

**Review by hugovk — COMMENTED** [review]
These files haven't been completely cleaned of _old_ docs warnings, so we don't let them warn on the CI, but now is a good time to prevent creating _new_ warnings:

```diff
diff --git a/Doc/whatsnew/2.6.rst b/Doc/whatsnew/2.6.rst
index 84bb651e68..7236e1cad7 100644
--- a/Doc/whatsnew/2.6.rst
+++ b/Doc/whatsnew/2.6.rst
@@ -2465,7 +2465,7 @@ changes, or look through the Subversion logs for all the details.
   (All changes contributed by Lars Gustäbel).
 
 * An optional ``timeout`` parameter was added to the
-  :class:`telnetlib.Telnet` class constructor, specifying a timeout
+  :class:`!telnetlib.Telnet` class constructor, specifying a timeout
   measured in seconds.  (Added by Facundo Batista.)
 
 * The :class:`tempfile.NamedTemporaryFile` class usually deletes
diff --git a/Doc/whatsnew/3.6.rst b/Doc/whatsnew/3.6.rst
index 3a681754e2..1cd33dfb60 100644
--- a/Doc/whatsnew/3.6.rst
+++ b/Doc/whatsnew/3.6.rst
@@ -1520,7 +1520,7 @@ rather than the version that is being emulated for the process
 telnetlib
 ---------
 
-:class:`~telnetlib.Telnet` is now a context manager (contributed by
+:class:`!telnetlib.Telnet` is now a context manager (contributed by
 Stéphane Wirtel in :issue:`25485`).
 
 
diff --git a/Misc/NEWS.d/3.9.0a1.rst b/Misc/NEWS.d/3.9.0a1.rst
index 0888a5c430..7da4385973 100644
--- a/Misc/NEWS.d/3.9.0a1.rst
+++ b/Misc/NEWS.d/3.9.0a1.rst
@@ -99,7 +99,7 @@ after whitespace, e.g. '127.0.0.1 whatever'.
 
 Adds audit events for :mod:`ensurepip`, :mod:`ftplib`, :mod:`glob`,
 :mod:`imaplib`, :mod:`nntplib`, :mod:`pdb`, :mod:`poplib`, :mod:`shutil`,
-:mod:`smtplib`, :mod:`sqlite3`, :mod:`subprocess`, :mod:`telnetlib`,
+:mod:`smtplib`, :mod:`sqlite3`, :mod:`subprocess`, :mod:`!telnetlib`,
 :mod:`tempfile` and :mod:`webbrowser`, as well as :func:`os.listdir`,
 :func:`os.scandir` and :func:`breakpoint`.
 
```

**Comment by vstinner** [comment]
I didn't know the ``!telnetlib`` syntax in Sphinx. I modified my PR to use this style and I added a reference to PEP 594.

**Comment by vstinner** [comment]
According to a search on ``import telnetlib|from telnetlib`` regex on PyPI top 5,000 projects (2023-04-13), the following projects use telnetlib.

Affected projects (15):
* ansible (7.4.0)
* bandit (1.7.5)
* bflb-iot-tool (1.8.4)
* boto (2.49.0)
* gevent (22.10.2)
* gsutil (5.23)
* homeassistant (2023.4.3)
* junos-eznc (2.6.7)
* mercurial (6.4.1)
* mobly (1.12.1)
* mozrunner (8.2.1)
* netmiko (4.1.2)
* pudb (2022.1.3)
* robotframework (6.0.2)
* ruff (0.0.261)

**Pull request merged**